File: day2/day2.py
horizontal = 0;
depth = 0;
aim = 0;

f = open("input.txt");
content = f.read().splitlines();

for line in content :
    splitted = line.split(' ');
    print(splitted);
    if splitted[0] == "forward":
        horizontal += int(splitted[1]);
        depth += aim * int(splitted[1]);
    elif splitted[0] == "up":
        aim -= int(splitted[1]);
    elif splitted[0] == "down":
        aim += int(splitted[1]);
    

print(horizontal*depth);

File: day2/day2.hs
{-# OPTIONS_GHC -Wno-incomplete-patterns #-}

main :: IO ()
main = do
    res <- foldl calculate (0,0,0) <$> (map words . lines <$> readFile "input.txt")
    print $ final res

calculate :: (Int, Int, Int) -> [String] -> (Int, Int, Int)
calculate (h,d,a) s = case head s of
    "forward" -> (h + value, d + (a * value), a)
    "up"      -> (h, d, a - value)
    "down"    -> (h, d, a + value)
    where value = read $ last s :: Int

final :: Num a => (a, a, a) -> a
final (h,d,a) = h*d